optimum/neuron/generation/utils.py

```diff
@@ -218,6 +223,442 @@ def _update_past(past_key_values):
 
         return model_kwargs
 
+    @staticmethod
+    def _expand_inputs_for_generation(
+        expand_size: int = 1,
+        is_encoder_decoder: bool = False,
+        input_ids: Optional[torch.LongTensor] = None,
+        **model_kwargs,
+    ) -> Tuple[torch.LongTensor, Dict[str, Any]]:
+        """Expands tensors from [batch_size, ...] to [batch_size * expand_size, ...]"""
+
+        def _expand_dict_for_generation(dict_to_expand):
+            for key in dict_to_expand:
+                if dict_to_expand[key] is not None and isinstance(dict_to_expand[key], torch.Tensor):
+                    if len(dict_to_expand[key].shape) == 2:
+                        dict_to_expand[key] = dict_to_expand[key].repeat(1, expand_size).view(-1, dict_to_expand[key].shape[1])
+                    elif len(dict_to_expand[key].shape) <= 1:
+                        dict_to_expand[key] = dict_to_expand[key].repeat(expand_size)
+                    else:
+                        dict_to_expand[key] = torch.concat([tensor.unsqueeze(0).repeat(expand_size,1,1) for tensor in dict_to_expand[key]])
+            return dict_to_expand
+
+        if input_ids is not None:
+            # Manual repeat interleave
+            input_ids = input_ids.repeat(1, expand_size).view(-1, input_ids.shape[1])
+
+        model_kwargs = _expand_dict_for_generation(model_kwargs)
+
+        if is_encoder_decoder:
+            if model_kwargs.get("encoder_outputs") is None:
+                raise ValueError("If `is_encoder_decoder` is True, make sure that `encoder_outputs` is defined.")
+            model_kwargs["encoder_outputs"] = _expand_dict_for_generation(model_kwargs["encoder_outputs"])
+
+        return input_ids, model_kwargs
+
+    def beam_search(
+        self,
+        input_ids: torch.LongTensor,
+        beam_scorer: BeamScorer,
+        logits_processor: Optional[LogitsProcessorList] = None,
+        stopping_criteria: Optional[StoppingCriteriaList] = None,
+        max_length: Optional[int] = None,
+        pad_token_id: Optional[int] = None,
+        eos_token_id: Optional[Union[int, List[int]]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_scores: Optional[bool] = None,
+        return_dict_in_generate: Optional[bool] = None,
+        synced_gpus: Optional[bool] = False,
+        seq_length: Optional[int] = int,
```

Comment:
Most likely a typo here.
```suggestion
        seq_length: Optional[int] = ?,
```